src/parser.rs

```diff
@@ -277,6 +279,11 @@ fn lexer() -> impl Parser<char, Vec<(Token, Span)>, Error = LexerError> {
         .then_ignore(just("i64"))
         .map(|n| Token::Int64(n as i64));
 
+    let int128 = integer
```

Comment:
`integer` currently parses to an i64. So this won't be able to cover the whole 128 bit range. Maybe it's enough to update `integer` to parse into an i128 instead.

Reply:
This seems to have worked! I also added i128 support to data sections for consistency even though I don't know why you wouldn't just use smaller units instead. I think I've managed to be consistent enough with using I128 to describe the literal type and V128 to describe the WASM type but I'm not 100% sure I got this right.